apps/client/src/features/rundown/event-editor/composite/EventTriggers.tsx

```diff
@@ -0,0 +1,159 @@
+import { IconButton, Select, Tooltip } from '@chakra-ui/react';
+import { TimerLifeCycle, timerLifecycleValues, TriggerDTO } from 'ontime-types';
+
+import useAutomationSettings from '../../../../common/hooks-query/useAutomationSettings';
+import * as Editor from '../../../editors/editor-utils/EditorUtils';
+import { EditorUpdateFields } from '../EventEditor';
+
+import style from '../EventEditor.module.scss';
+
+type EventTriggers = Record<string, TriggerDTO>;
+
+interface EventTriggersProps {
+  triggers?: EventTriggers;
+  handleSubmit: (field: EditorUpdateFields, value: string) => void;
+}
+
+export default function EventTriggers(props: EventTriggersProps) {
+  const { triggers, handleSubmit } = props;
+
+  return (
+    <>
+      {triggers !== undefined && <ExistingEventTriggers triggers={triggers} handleSubmit={handleSubmit} />}
+      <EventTriggerForm handleSubmit={handleSubmit} triggers={triggers} />
+    </>
+  );
+}
+
+interface EventTriggerFormProps {
+  handleSubmit: (field: EditorUpdateFields, value: string) => void;
+  triggers?: EventTriggers;
+}
+
+function EventTriggerForm(props: EventTriggerFormProps) {
+  const { handleSubmit, triggers } = props;
+  const { data: automationSettings } = useAutomationSettings();
+  const [automationId, setAutomationId] = useState<string | undefined>(undefined);
+  const [cycleValue, setCycleValue] = useState(TimerLifeCycle.onStart);
+
+  const isInvalidTrigger = useCallback(
+    (cycle: TimerLifeCycle, automationId?: string): false | string => {
+      if (automationId === undefined) {
+        return 'Select an automation';
+      }
+      if (!Object.keys(automationSettings.automations).includes(automationId)) {
+        return 'This automation dose not exist';
+      }
+      if (triggers === undefined) {
+        return false;
+      }
+      return Object.values(triggers).some((t) => t.automationId === automationId && t.trigger === cycle)
+        ? 'Automation can only be used once'
+        : false;
+    },
+    [automationSettings.automations, triggers],
+  );
+
+  return (
+    <div className={style.inline}>
+      <Select
+        size='sm'
+        variant='ontime'
+        value={automationId}
+        defaultValue='default'
+        onChange={(e) => setAutomationId(e.target.value)}
+      >
+        <option disabled value='default'>
+          select an automation
+        </option>
+        {Object.values(automationSettings.automations).map(({ id, title }) => (
+          <option key={id} value={id}>
+            {title}
+          </option>
+        ))}
+      </Select>
+      <Select
+        size='sm'
+        variant='ontime'
+        value={cycleValue}
+        onChange={(e) => setCycleValue(e.target.value as TimerLifeCycle)}
+        defaultValue={TimerLifeCycle.onStart}
+      >
+        {/* Notice here that we don't support onStop */}
+        {['onLoad', 'onStart', 'onPause', 'onFinish', 'onWarning', 'onDanger'].map((cycle) => (
```

Comment:
This sounds like a decision we should extract from the view. In general, views shouldnt contain many decisions other than presentation ones

Reply:
I usually just make a auxiliary file like event trigger.constants.ts